feat: added a new bot deployment detection bot (#1)
gautam2002/forta-bots-challenges: detection-bot/README.md

# New Bot Deployment Detection Bot

## Description

> This bot detects if a new bot is deployed from a specific address. It achieves this by filtering the transactions that match the createAgent function signature.

Reviewer: ```suggestion
This bot detects if a new bot is deployed from a specific address. It achieves this by filtering the transactions that match the `createAgent` function signature.
```

## Supported Chains

- Polygon (Had to polygon `jsonRpcUrl` to test out locally)

## Alerts

- NEMD-1
  - Fired when a new bot is created by the Nethermind address(deployer).
  - Severity is set to "info".
  - Type is always set to "info".
  - Metadata:
    - `owner`: Owner Address
    - `agentId`: Id of the Deployed Bot

>     - `Ipfsmetadata`: Bot metadata hash ID stored in IPFS

Reviewer: ```suggestion
    - `IpfsMetadata`: Bot metadata hash ID stored in IPFS
```

Using camelCase (should also be changed in the bot logic)

    - `chainIds`: List of supported Chains

## Test Data

> The agent behavior can be verified with the following transactions:

Reviewer: ```suggestion
The bot behavior can be verified with the following transactions:
```

> # New Bot Deployment Detection Agent

Reviewer 2: ```suggestion
# New Bot Deployment Detection Bot
```

## Description

> This agent detects if a new bot is deployed from a specific address. It achieves this by filtering the transactions that match the createAgent function signature.

Reviewer 2: ```suggestion
This bot detects if a new bot is deployed from a specific address. It achieves this by filtering the transactions that match the createAgent function signature.
```

## Supported Chains

- Polygon (Had to polygon `jsonRpcUrl` to test out locally)

## Alerts

> Describe each of the types of alerts fired by this agent

Reviewer 2: ```suggestion

```

- NEMD-1
  - Fired when a new bot is created by the Nethermind address(deployer).
  - Severity is set to "info".
  - Type is always set to "info".

>   - Metadata: `owner`(owner address), `Ipfsmetadata`(Bot metadata content hash ID IPFS), `agentID`(uint256), `chainIds`

Reviewer 2: ```suggestion
  - Metadata: 
          - `owner`: Owner address, 
          - `Ipfsmetadata`: Bot metadata content hash ID IPFS, 
          - `agentID`: Id of the deployed bot, 
          - `chainIds`: List of supported chains
```

## Test Data

The agent behavior can be verified with the following transactions:

- [0xb9fffc84511a7135aeed7db1f560951dac6c9ea49c8de353e9c41d70a322b210](https://polygonscan.com/tx/0xb9fffc84511a7135aeed7db1f560951dac6c9ea49c8de353e9c41d70a322b210)

The response looks like this locally when

```
npm run tx 0xb9fffc84511a7135aeed7db1f560951dac6c9ea49c8de353e9c41d70a322b210

1 findings for transaction 0xb9fffc84511a7135aeed7db1f560951dac6c9ea49c8de353e9c41d70a322b210 {
  "name": "Nethermind Bot Detection",
  "description": "Detects every new Bot deployed from Nethermind Address",
  "alertId": "Nemd-1",
  "protocol": "ethereum",
  "severity": "Info",
  "type": "Info",
  "metadata": {
    "owner": "0x88dC3a2284FA62e0027d6D6B1fCfDd2141a143b8",
    "Ipfsmetadata": "QmPqj6HNP..........",
    "agentId": "880547971935321473673210414.....",
    "chainIds": "1"
  },
  "addresses": []

> }

Reviewer 2: this block is not necessary because there could be many test cases.
